### PR TITLE
Add a custom combineReducers option

### PR DESCRIPTION
**Pull request by abettadapur**
When using immutableJS with Redux, a custom combineReducer must be provided.
This change allows the consumer to specify a custom combine function

### CODE REVIEW

File: packages/redux-dynamic-modules-core/src/Managers/ModuleManager.ts [hunk 1]
```diff
@@ -19,7 +19,10 @@ export interface IModuleManager<State> extends IItemManager<IModule<State>> {
 
 export function getModuleManager<State>(
     middlewareManager: IItemManager<Middleware>,
-    extensions: IExtension[]
+    extensions: IExtension[],
+    reducerCombiner?: (
+        reducers: ReducersMapObject<State, any>
```
Comment (navneet-g) [inline packages/redux-dynamic-modules-core/src/Managers/ModuleManager.ts]: Please add some doc comments, on usecase and usage pattern.

File: packages/redux-dynamic-modules-core/src/Managers/ModuleManager.ts [hunk 2]
```diff
@@ -19,7 +19,10 @@ export interface IModuleManager<State> extends IItemManager<IModule<State>> {
 
 export function getModuleManager<State>(
     middlewareManager: IItemManager<Middleware>,
-    extensions: IExtension[]
+    extensions: IExtension[],
+    reducerCombiner?: (
```
Comment (navneet-g) [inline packages/redux-dynamic-modules-core/src/Managers/ModuleManager.ts]: customCombineReducer?

**Comment by abettadapur** [comment]
@codeocelot So sorry for the delay here, been very busy with my main job

I will get this published for you tonight

File: packages/redux-dynamic-modules-core/src/Managers/ModuleManager.ts (hunk 2 above)
Comment (navneet-g) [reply packages/redux-dynamic-modules-core/src/Managers/ModuleManager.ts]: or advanceCombineReducer.

**Comment by abettadapur** [comment]
There is a TS error here I need to fix and I will add tests

**Comment by abettadapur** [comment]
Test Plan:
All tests pass
All the examples continue to work, after updating their createStore functions

**Pull request merged**